File: ead_clustering/xml_parser.py
import xml.etree.ElementTree as ET
import re
from pathlib import Path
from typing import List

NAMESPACE_REGEX = re.compile(r'\{[^}]+\}')

def extract_paths_from_element(element, current_path="") -> List[str]:
    """Récursivement, extrait tous les chemins à partir d’un élément XML."""
    clean_tag = NAMESPACE_REGEX.sub('', element.tag)
    new_path = f"{current_path}/{clean_tag}" if current_path else clean_tag
    paths = [new_path]
    for child in element:
        paths.extend(extract_paths_from_element(child, new_path))
    return paths

def extract_xml_paths(xml_file: Path) -> List[str]:
    """
    Parse un fichier XML et retourne une liste des chemins de balises (structure).
    """
    try:
        tree = ET.parse(xml_file)
        root = tree.getroot()
        return extract_paths_from_element(root)
    except ET.ParseError as e:
        print(f"[ERREUR] Parsing XML échoué ({xml_file.name}) : {e}")
        return []

File: julia_version/xml_parser.jl
using LightXML

function read_xml(xml_file::String)::Vector{String}
    # Bon exemple pour apprendre à lire un XML !
    tree = nothing
    paths = String[]

    try
        tree = parse_file(xml_file)
        root_element = root(tree)
        println(root_element)
        free(tree)
    catch e
        println("Erreur de parsin XML / ($xml_file) : ", e)
    finally
        if tree !== nothing
            free(tree)
        end
    end
    return paths
end

function extract_paths_from_element(element::XMLElement, current_path=""::String)::Vector{String}
    clean_tag = replace(name(element), r"\{[^}]+\}" => "")
    new_path = isempty(current_path) ? clean_tag : current_path * "/" * clean_tag
    paths = Vector{String}()
    push!(paths, new_path)
    for child in child_elements(element)
        append!(paths, extract_paths_from_element(child, new_path))
    end
    return paths
end

function extract_xml_paths(xml_file::String)::Vector{String}
    tree = nothing
    paths = String[]

    try
        tree = parse_file(xml_file)
        root_element = root(tree)
        append!(paths, extract_paths_from_element(root_element))
    catch e
        println("Erreur de parsin XML / ($xml_file) : ", e)
    finally
        if tree !== nothing
            free(tree)
        end
    end
    return paths
end

paths = extract_xml_paths("ead_files/bnf.xml")
println(paths)